src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -1292,27 +1292,28 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ShouldBeValidatedAsExecutableReference>false</ShouldBeValidatedAsExecutableReference>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(IsTestProject)' == 'true' And '$(ValidateExecutableReferencesMatchSelfContained)' == ''">
-    <!-- Don't generate an error if a test project references a self-contained Exe.  Test projects
-         use an OutputType of Exe but will usually call APIs in a referenced Exe rather than try
-         to run it. -->
-    <ValidateExecutableReferencesMatchSelfContained>false</ValidateExecutableReferencesMatchSelfContained>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(IsTestProject)' == 'true'">
-    <!-- Don't generate an error if an Exe project references a test project. -->
-    <ShouldBeValidatedAsExecutableReference>false</ShouldBeValidatedAsExecutableReference>
-  </PropertyGroup>
-
-
   <UsingTask TaskName="ValidateExecutableReferences" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
   <PropertyGroup>
     <_UseAttributeForTargetFrameworkInfoPropertyNames Condition="$([MSBuild]::VersionGreaterThanOrEquals($(MSBuildVersion), '17.0'))">true</_UseAttributeForTargetFrameworkInfoPropertyNames>
   </PropertyGroup>
 
+  <Target Name="_CalculateIsVSTest">
+    <PropertyGroup>
+      <_IsVSTest Condition="'$(IsTestProject)' == 'true' and '$(IsTestingPlatformApplication)' != 'true'">true</_IsVSTest>
```

Comment:
TODO: Rename to `_IsVSTestTestProject` once this is reviewed and we are okay with the change.